topic_browser_plugin/processing.go

```
@@ -0,0 +1,44 @@
+
+	// Buffer original message (for ACK)
+	t.messageBuffer = append(t.messageBuffer, msg)
+
+	// Add to per-topic ring buffer
+	t.addEventToTopicBuffer(topic, event)
+
+	// Update fullTopicMap with cumulative metadata and emit complete state once per interval
+	// This maintains the authoritative topic state with merged metadata across all messages
+	cumulativeMetadata := t.mergeTopicHeaders(unsTreeId, []*TopicInfo{topicInfo})
+	// Update topic info with cumulative metadata before storing
+	topicInfoWithCumulative := *topicInfo // shallow copy
```

golangci-lint (1.64.8)

copylocks: assignment copies lock value to topicInfoWithCumulative: github.com/united-manufacturing-hub/benthos-umh/topic_browser_plugin.TopicInfo contains google.golang.org/protobuf/runtime/protoimpl.MessageState contains sync.Mutex

(govet)